### PR TITLE
Feature/admin undo actions

### PR DESCRIPTION
**Pull request by ppmg**
Admin actions to undo team deletion and user banning have been added.

### CODE REVIEW

File: app/src/main/kotlin/team/finder/api/admin/AdminController.kt [hunk 1]
```diff
@@ -25,44 +26,51 @@ class AdminController(
...
         val user = usersService.getUser(userDetails.discordId)
-        if (user == null || !user.isAdmin) {
-            return ResponseEntity(HttpStatus.NOT_FOUND)
-        }
+        return if (user?.isAdmin == true) user else null
+    }
+
+    @GetMapping("/admin/reports")
+    fun reports(): ResponseEntity<Any> {
+        val adminUser = getAuthorisedUser() ?: return ResponseEntity(HttpStatus.NOT_FOUND)
```
Comment (dallonf) [inline app/src/main/kotlin/team/finder/api/admin/AdminController.kt:37]: `?:` 😮 

Kotlin's pretty cool

File: app/src/main/kotlin/team/finder/api/admin/AdminController.kt [hunk 2]
```diff
@@ -85,6 +93,27 @@ class AdminController(
...
+
+        val userToRedeem = usersService.getUser(discordIdOfUserToRedeem) ?: return ResponseEntity(HttpStatus.NOT_FOUND)
+
+        userToRedeem.isBanned = false
+        usersService.saveUser(userToRedeem)
+
+        logger.info("[${ADMIN_TAG}] ${adminUser.name} has redeemed User ${userToRedeem.discordId}")
+
+        // @todo should we always re-instate the team automatically? Maybe they were banned for a bad team but are being given a second chance (in which case we shouldn't reinstate the team)
```
Comment (dallonf) [inline app/src/main/kotlin/team/finder/api/admin/AdminController.kt:107]: How about a query param option?

Comment (Willdotwhite) [reply app/src/main/kotlin/team/finder/api/admin/AdminController.kt:107]: I think a query param is definitely the way to go, although it wouldn't be the worst thing to say "your account is back, please make a nicer team this time" 

Getting the last team would be something like this, I think:

```sql
SELECT * FROM team t WHERE t.deleted_at IS NOT NULL ORDER BY t.deleted_at DESC LIMIT 1
```

File: README.md [hunk 3]
```diff
@@ -75,6 +75,45 @@ This example uses Docker, because setting up MySQL natively is a pain.
 
 You don't need to create the `teams` table in the DB for some reason - Java/SB magic allows the data to persist somehow.
 
+### Endpoints
```
Comment (Willdotwhite) [inline README.md:78]: Thanks so much for adding these 👍

**Review by Willdotwhite — APPROVED** [review]

**Pull request merged**